sentinel-adapter/sentinel-spring-webmvc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/spring/webmvc/SentinelInterceptor.java

```diff
@@ -0,0 +1,134 @@
+import com.alibaba.csp.sentinel.SphU;
+import com.alibaba.csp.sentinel.Tracer;
+import com.alibaba.csp.sentinel.adapter.spring.webmvc.callback.RequestOriginParser;
+import com.alibaba.csp.sentinel.adapter.spring.webmvc.config.SentinelSpringMvcConfig;
+import com.alibaba.csp.sentinel.adapter.spring.webmvc.callback.UrlCleaner;
+import com.alibaba.csp.sentinel.context.ContextUtil;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.alibaba.csp.sentinel.util.StringUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.servlet.HandlerMapping;
+
+/**
+ * Spring mvc interceptor that integrates with sentinel.
+ *
+ * @Author kaizi2009
+ */
+public class SentinelInterceptor extends AbstractSentinelInterceptor {
+    private static final String EMPTY_ORIGIN = "";
+    private final static String COLON = ":";
+    private SentinelSpringMvcConfig config;
+
+    public SentinelInterceptor(SentinelSpringMvcConfig config) {
+        if (config == null) {
+            throw new SentinelSpringMvcException("Config can not be null");
+        }
+        this.config = config;
+    }
+
+    public SentinelInterceptor() {
+        this(new SentinelSpringMvcConfig());
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+        Entry urlEntry = null;
+        Entry httpMethodUrlEntry = null;
+
+        try {
+            String resourceName = getResourceName(request);
+
+            UrlCleaner urlCleaner = config.getUrlCleaner();
+            if (urlCleaner != null) {
+                resourceName = urlCleaner.clean(resourceName);
+            }
+
+            if (!StringUtil.isEmpty(resourceName)) {
+                // Parse the request origin using registered origin parser.
+                String origin = parseOrigin(request);
+                ContextUtil.enter(SPRING_MVC_CONTEXT_NAME, origin);
+                urlEntry = SphU.entry(resourceName, EntryType.IN);
+                // Add method specification if necessary
+                if (config.isHttpMethodSpecify()) {
+                    httpMethodUrlEntry = SphU.entry(request.getMethod().toUpperCase() + COLON + resourceName,
+                            EntryType.IN);
+                }
+            }
+            final EntryContainer entryContainer = new EntryContainer().setUrlEntry(urlEntry)
+                    .setHttpMethodUrlEntry(httpMethodUrlEntry);
+            setEntryContainerInReqeust(request, config.getRequestAttributeName(), entryContainer);
+            return true;
+        } catch (BlockException e) {
+            //Throw BlockException and handle it in spring MVC
+            throw e;
+        } catch (RuntimeException e1) {
+            Tracer.traceEntry(e1, urlEntry);
+            Tracer.traceEntry(e1, httpMethodUrlEntry);
+            throw e1;
```

Comment:
What will cause this RuntimeException? and can ignore this RuntimeException here?

Reply:
Increment exception count to provided entry, see https://github.com/alibaba/Sentinel/wiki/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8#%E4%B8%9A%E5%8A%A1%E5%BC%82%E5%B8%B8%E7%BB%9F%E8%AE%A1-tracer